### PR TITLE
PYTHON-4506 Add autogen

### PR DESCRIPTION
**Pull request by caseyclements**
Adding work done together with @ranfysvalle02 (@Fabian) on the [Autogen](https://microsoft.github.io/autogen/)  framework.

As usual, we are pointing to the feature branch until it is merged. The upstream maintainers have been very responsive.

### CODE REVIEW

File: autogen/run.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+$PYTHON_BINARY -m venv venv
+source venv/bin/activate
+#  Install autogen with extras
+$PYTHON_BINARY -m pip install .[test,"retrievechat-mongodb"]
+
+
+# Run tests. Sensitive variables in Evergreen come from Evergreen project: ai-ml-pipeline-testing/
+MONGODB_URI=$autogen_mongodb_uri \
+MONGODB_DATABASE="autogen_test_db" \
+py.test -v test/agentchat/contrib/vectordb/test_mongodb.py
```
Comment (Jibola) [inline autogen/run.sh:23]: As an aside: 

A lot of this set-up seems easily reproducible. Is it at all worth it to consolidate this script into a singular function?

**Review by Jibola — APPROVED** [review]
LGTM!

Having the feature branch is a great addition. :)

**Review by Jibola — CHANGES_REQUESTED** [review]
Just saw this error

```
[2024/06/24 14:03:37.639] + py.test -v test/agentchat/contrib/vectordb/test_mongodb.py
[2024/06/24 14:03:37.639] ../run.sh: line 21: py.test: command not found
```

File: autogen/run.sh (hunk 1 above)
Comment (caseyclements) [reply autogen/run.sh]: Nah. It's simple and clean and everyone hates writing bash. :)

**Review by Jibola — APPROVED** [review]

**Pull request merged**